pilot/helpers/agents/CodeMonkey.py

```diff
@@ -87,32 +90,20 @@ def implement_code_changes(
         file_name, file_content = self.get_original_file(code_changes_description, step, files)
         content = file_content
 
-        # If the file is non-empty and larger than the threshold, attempt to replace individual code blocks
-        if file_content and len(file_content.splitlines()) > self.SMART_REPLACE_THRESHOLD:
-            replace_complete_file, content = self.replace_code_blocks(
-                step,
-                convo,
-                standalone,
-                code_changes_description,
-                file_content,
-                file_name,
-                files,
-            )
-        else:
-            # Just replace the entire file
-            replace_complete_file = True
-
-        # If this is a new file or replacing individual code blocks failed,
-        # replace the complete file.
-        if replace_complete_file:
-            content = self.replace_complete_file(
-                convo,
-                standalone,
-                code_changes_description,
-                file_content,
-                file_name, files
-            )
+        # Get the new version of the file
+        content = self.replace_complete_file(
+            convo,
+            standalone,
+            code_changes_description,
+            file_content,
+            file_name, files
+        )
 
+        # Review the changes and only apply changes that are useful/approved
+        if content and content != file_content:
+            content = self.review_change(convo, code_changes_description, file_name, file_content, content)
+
+        # If we have changes, update the file
         if content and content != file_content:
```

Comment:
here i think we have to check if content == file_content and if yes (we didn't apply any changes) we should retry step with new changes. This could be a bit too complex for this PR so we can make new task for that